aws/aws.go

```diff
@@ -197,6 +214,26 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 			endpoint = value
 		case "profile":
 			opts = append(opts, awsv2cfg.WithSharedConfigProfile(value))
+		case "dualstack":
+			dualStackParam, err := strconv.ParseBool(value)
+			if err != nil {
+				return awsv2.Config{}, fmt.Errorf("invalid value for dualstack: %w", err)
+			}
+			dualStack := awsv2.DualStackEndpointStateDisabled
+			if dualStackParam {
+				dualStack = awsv2.DualStackEndpointStateEnabled
+			}
+			opts = append(opts, awsv2cfg.WithUseDualStackEndpoint(dualStack))
+		case "fips":
+			fipsParam, err := strconv.ParseBool(value)
+			if err != nil {
+				return awsv2.Config{}, fmt.Errorf("invalid value for fips: %w", err)
+			}
+			fips := awsv2.FIPSEndpointStateDisabled
+			if fipsParam {
+				fips = awsv2.FIPSEndpointStateEnabled
+			}
```

Comment:
For hostnameImmutable, we kept it so that it could be added to the customResolver below, should we do that for these 2 as well?

Reply:
Sure, how does this look now? This will probably change with the move to service-specifc endpoint resolvers.